### PR TITLE
Adds analytics for upgraded & canceled subscriptions

### PR DESCRIPTION
**Pull request by discorick**
`/settings/{owner}/upgrade` when Upgrade is clicked (popping Stripe modal) is blocked by #158/#189/#209

### CODE REVIEW

**Comment by dahlbyk** [comment]
Random observation: `/settings/profile` doesn't have a payload for Identify, just the ID. Probably fine since we associate info with that ID elsewhere, but wanted to point it out since it surprised me.
image

**Comment by discorick** [comment]
Is that the JS or Ruby call @dahlbyk ?

File: vendor/engines/saas/app/jobs/saas/plans_destroy_job.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Saas
+  class PlansDestroyJob < ActiveJob::Base
+
+    def perform(params)
+      if params[:event] == 'plan_canceled'
+        params['url'] = "/settings/#{params[:repo_owner]}/canceled"
```
Comment (dahlbyk) [inline vendor/engines/saas/app/jobs/saas/plans_destroy_job.rb]: This should be `canceling`; `canceled` is for hits on a board that has already been canceled.

**Comment by dahlbyk** [comment]
> Is that the JS or Ruby call @dahlbyk ?

Both, actually.
image

File: vendor/engines/saas/app/jobs/saas/plans_destroy_job.rb (hunk 1 above)
Comment (discorick) [inline vendor/engines/saas/app/jobs/saas/plans_destroy_job.rb]: Your right, I mixed up on the verbiage

**Comment by dahlbyk** [comment]
Looking good.
image

**Pull request merged**